src/libbson/tests/test-string.c

```diff
@@ -303,6 +304,68 @@ test_bson_strcasecmp (void)
    BSON_ASSERT (bson_strcasecmp ("FoZ", "foo") > 0);
 }
 
+static void
+test_bson_string_capacity (void *unused)
+{
+   BSON_UNUSED (unused);
+
+   char *large_str = bson_malloc (UINT32_MAX);
+   memset (large_str, 's', UINT32_MAX); // Do not NULL terminate. Each test case sets NULL byte.
+
+   // Test the largest possible string that can be constructed.
+   {
+      large_str[UINT32_MAX - 1] = '\0'; // Set size.
```

Comment:
```suggestion
      large_str[UINT32_MAX - 1u] = '\0'; // Set size.
```
Integer literal sign consistency.

Also applies to other arithmetic expressions in this file.